tardis/visualization/tools/sdec_plot.py

```diff
@@ -568,6 +576,11 @@ def _calculate_plotting_data(
                 "allowed values are 'virtual' or 'real'"
             )
 
+        if packets_mode == "virtual" and self.data[packets_mode] is None:
+            raise ValueError(
+                'SDECPlotter doesn\'t have any data for virtual packets population and SDEC plot for the same was requested.\nEither set virtual_packet_logging: True in your configuration file to generate SDEC plot with virtual packets, or pass packets_mode="real" in your function call to generate SDEC plot with real packets.'
```

Comment:
It's better to break long string to multiple lines following 80 char limit. Also if you use `"..."` for your string you will not need to escape `'` used inside your string - see how L575-L576 does this